NBitcoin/Transaction.cs

```diff
@@ -200,10 +193,32 @@ public override string ToString()
 		{
 			return $"{Hash}-{N}";
 		}
+
+		public int CompareTo(object obj)
+			=> CompareTo(obj as OutPoint);
+		public int CompareTo(OutPoint other)
+		{
+			if (other is null || this < other)
+				return -1;
+			if (this == other)
+				return 0;
+			return 1;
+		}
+
+		public override bool Equals(object obj)
+			=> Equals(obj as OutPoint);
+
+		public bool Equals(OutPoint other)
+		{
+			if (other is null)
+				return false;
+			return other == this;
+		}
+
 	}
 
 
-	public class TxIn : IBitcoinSerializable
+	public class TxIn : IBitcoinSerializable, IEquatable<TxIn>

```

Comment:
`IComparable` makes no sense for `TxIn` outpoint, unless you want to implement @kristovatla's BIP69: https://github.com/bitcoin/bips/blob/master/bip-0069.mediawiki

Reply:
NACK on this one, `TxIn` can be subclassed by shitcoins so you can't assume you know equality.